packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/EsxiCredentialsEdit.tsx

```diff
@@ -22,9 +22,9 @@ import { EditComponentProps } from '../BaseCredentialsSection';
 export const EsxiCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onChange }) => {
   const { t } = useForkliftTranslation();
 
-  const user = safeBase64Decode(secret?.data?.user || '');
+  const user = safeBase64Decode(secret?.data?.user);
+  const password = safeBase64Decode(secret?.data?.password);
   const url = safeBase64Decode(secret?.data?.url || '');
```

Comment:
If we are doing this refactor of using `undefined` as a valid value, we should remove the `|| ''` from all fields in the file we touch in this PR (rest we can do the bigger refactor)

Reply:
The fields that I changed as part of this PR are the ones that must use 'undefined' for the init state since you didn't want to change the 'error' validation state if they are set to empty. Not all fields require supporting `undefined` as a legitimate validation state. Supporting it requires changing the `FieldValidators` for all those fields as well.

For the URL - it's already supported so can be changed.

For all other fields,  I suggest doing that as part of https://github.com/kubev2v/forklift-console-plugin/issues/1059. 
WDYT?

Reply:
ok :+1: